js.go

```diff
@@ -1844,7 +1844,14 @@ func (sub *Subscription) activityCheck() {
 	}
 
 	active := jsi.active
-	jsi.hbc.Reset(jsi.hbi)
+	// the timer could still have gone off even though it was stopped in removeSub,
+	// in this case it will be nil because removeSub sets it to nil
+	if jsi.hbc != nil {
+		jsi.hbc.Reset(jsi.hbi)
+	} else {
+		active = false
```

Comment:
Not 100% sure whether we should be setting `active` or `closed` to false here in this dangling execution